### ISSUE TITLE
Some songs are not playing!

### ISSUE THREAD

**Issue by Am10101578** [vfsfitvnm/ViMusic#1292]
### Steps to reproduce the bug

I was listening to an album by Lana Del Rey and some songs are not playing !!!!

### Expected behavior

Please let those songs play man.

### Actual behavior

Those songs were not playing...the timing of the song was also not displaying.

### ViMusic version

Latest one

### Android version

Android 12

**Comment by twistios** [comment]
Please add some more information to your bug report, i.e. at least which album, what songs.
For example I tested the "Born to Die"-Album and every song was playing correctly.

Info:
- Android 13
- ViMusic v0.5.4

**Comment by Am10101578** [comment]
My bad! The song: F*ck it I love you and NFR (title track) in the album Norman f Rockwell album.
Then, the song F*cked my way up to the top from the ultraviolence album, along with dealer and fishtail from Blue banisters and Ocean Boulevard Albums respectively. These songs won't play.
I only noticed these yet..but I fear there may be more with other artists...
I really appreciate this application and would like for these bugs to be dealt with.
Thank you! )

**Comment by Leleo98** [comment]
Albums by certain artists, such as Charlie Puth's Voicenotes, do not appear in the player

**Comment by Leleo98** [comment]
Albums by certain artists, such as Charlie Puth's Voicenotes, do not appear in the player

**Comment by twistios** [comment]
I tested "F it I love you", NFR and "F my way up to the top" and I was able to reproduce the problem on my device.
It seems to be related to https://github.com/vfsfitvnm/ViMusic/issues/1222.

**Comment by Am10101578** [comment]
Agreed, it shows "unexpected playback error" what is the solution, if there is any?

**Comment by twistios** [comment]
I am not a maintainer of this Project so I can only guess about this. It could be that this is intended behavior as there is currently no age-check in the app so that maybe this songs are not playing because it defaults to only unrestricted content.

**Comment by Krishnaraturi** [comment]
Nro listen im a fan of xxxtentacion and his songs are not playing like save me, Jocelyn flores, f*ck love and many more plz fix it

**Comment by mr-ema** [comment]
Edit: This does not fix all songs; some of them require logging in to be played, such as the one in issue #1352.

I believe there might be an issue with the YT API. In the browser the same happens the `POST` request fails when there's an age restriction and typically, YT displays a dialog to handle such restrictions.  After pressing the button within the dialog, the same request is made, but with an additional field.

## Here's A Breakdown

</br>

Both the requests with and without age restrictions share the same structure:

```
{
  captionParams: {},
  context: { ... },
  cpn: "...",
  playbackContext: { ... },
  serviceIntegrityDimensions: { ... },
  videoId: "yQFBKx6eq8s"
}
```

</br>

However, the response differs:

- Requests without age restrictions receive a [200] response with a `playabilityStatus` object showing `status: "OK"`.
- Requests with age restrictions receive a [200] response with a `playabilityStatus` object showing `status: "CONTENT_CHECK_REQUIRED"`, This might cause the app to break since there is no statement to handle such error status in `ViMusic/innertube/src/main/kotlin/it/vfsfitvnm/innertube/requests/Player.kt`.

```
{
  responseContext: Object { maxAgeSeconds: 0, serviceTrackingParams: […] },
  playabilityStatus: Object { status: "OK", ... }
  /* ... */
}

{
  responseContext: Object { maxAgeSeconds: 0, serviceTrackingParams: […] },
  playabilityStatus: Object { status: "CONTENT_CHECK_REQUIRED", ... }
  /* ... */
}
```

</br>

In YT to handle the error status a dialog is displayed and if accepted, the same request is remade with an extra field:

```
{
  captionParams: {},

  contentCheckOk: true, /* This one */

  context: { ... },
  cpn: "...",
  playbackContext: { ... },
  racyCheckOk: true,
  serviceIntegrityDimensions: { ... },
  videoId: "jpBYWpKKgTc"
}
```

</br>
</br>

Finally here's a simple workaround to bypass the dialog. This could serve as a starting point for implementing a more sophisticated solution. I'm loving the app and I would fix it myself but I don't have any Kotlin experience for the moment.

```
diff --git a/innertube/src/main/kotlin/it/vfsfitvnm/innertube/models/bodies/PlayerBody.kt b/innertube/src/main/kotlin/it/vfsfitvnm/innertube/models/bodies/PlayerBody.kt
index b09dd2f..0327158 100644
--- a/innertube/src/main/kotlin/it/vfsfitvnm/innertube/models/bodies/PlayerBody.kt
+++ b/innertube/src/main/kotlin/it/vfsfitvnm/innertube/models/bodies/PlayerBody.kt
@@ -5,6 +5,7 @@ import kotlinx.serialization.Serializable
 
 @Serializable
 data class PlayerBody(
+    val contentCheckOk: Boolean = true,
     val context: Context = Context.DefaultAndroid,
     val videoId: String,
     val playlistId: String? = null
```
